autoscheduler/frontend/src/utils/throttle.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Function to throttles the execution of other functions
```

Comment:
I think this is due for more comments - the starting syntax of this is _weird_.

So from how I understand it (just spit-balling to double check my understanding):, `throttle` is a function that calls itself immediately and then in turn returns a function (represented by the inner `return (id: string, ....`). So the first 3 lines in it (defining the variables, e.g. `const interval`) is only called _once_, but the inside of it is called whenever you call the function (like in `CourseSelectColumn`). Javascript is so weird

Reply:
Yeah, the technical term is an "immediately invoked function expression", and I used it for the reason you described (keeping an internal state for the function). When the module is imported, it immediately creates the data structures used to keep track of functions, and calling it from somewhere will use the function that's returned by it.